pkg/scheduler/scheduler.go

```diff
@@ -281,10 +286,33 @@ func New(client clientset.Interface,
 	sched.StopEverything = stopEverything
 	sched.client = client
 
-	addAllEventHandlers(sched, informerFactory)
+	// Build dynamic client and dynamic informer factory
+	var dynInformerFactory dynamicinformer.DynamicSharedInformerFactory
```

Comment:
Good point. I need to think about it. I'm a bit concerned that would add extra burden for plugin developers, such as fetching the GVR before add event handlers, etc. Probably need to compare it with how controller-runtime obstacles similar problems, and come up with a better abstraction.

Reply:
I rethink it a bit. It's better not to enforce out-of-tree plugins to use dynamic informers as typed ones are more applicable and easy to manipulate, in both core logic and tests.

Reply:
We are not forcing them to use it, but I think it is useful to make it available to them since the cache is already populated for the types they registered events for. If they create there own informer for the same type, aren't we wasting memory because we will have two informerFactor caches?

Reply:
> but I think it is useful to make it available to them since the cache is already populated for the types they registered events for

Not quite sure the plugin developers would really use it, as the typed informer and other generated stuff are way easier to manipulate.

> If they create their own informer for the same type, aren't we wasting memory because we will have two informerFactory caches?

Yes, this is true.